mytoyota/controller.py

```diff
@@ -212,38 +230,36 @@ async def request(  # pylint: disable=too-many-branches
                 HTTPStatus.OK,
                 HTTPStatus.ACCEPTED,
             ]:
-                result = response.json()
-            elif response.status_code == HTTPStatus.NO_CONTENT:
-                # This prevents raising or logging an error
-                # if the user have not setup Connected Services
-                result = None
-                _LOGGER.debug("Connected services is disabled")
-            elif response.status_code == HTTPStatus.INTERNAL_SERVER_ERROR:
-                response = response.json()
-                if "code" in response:
-                    error = ToyotaApiError(
-                        "Internal server error occurred! Code: "
-                        + response.get("code")
-                        + " - "
-                        + response.get("message"),
-                    )
-                else:
-                    error = ToyotaApiError(
-                        "Internal server error occurred! - " + response
-                    )
+                ret: dict[str, Any] = response.json()
+                if "payload" in ret:
+                    return ret["payload"]
+
+                return ret
+
+            # Errored if we get here
+            # pp.pprint(response.request.method)
+            # pp.pprint(response.request.url)
+            # pp.pprint(response.request.headers)
+            # pp.pprint(response.request.content)
 
-                raise error
+            if response.status_code == HTTPStatus.NO_CONTENT:
```

Comment:
this looks a bit cluttery in my opinion.
I think the error handling here can be improved by abstracting it into a dictionary lookup:

```python
            if response.status_code in [HTTPStatus.OK, HTTPStatus.ACCEPTED]:
                ret = response.json()
                return ret.get("payload", ret)

            error_handling = {
                HTTPStatus.NO_CONTENT: "NO_CONTENT",
                HTTPStatus.INTERNAL_SERVER_ERROR: "INTERNAL_SERVER_ERROR",
                HTTPStatus.BAD_GATEWAY: "Servers are overloaded, try again later",
                HTTPStatus.SERVICE_UNAVAILABLE: "Servers are temporarily unavailable",
                HTTPStatus.FORBIDDEN: "Action is not supported on this vehicle",
            }
            error_message = error_handling.get(response.status_code, f"HTTP: {response.status_code} - {response.text}")
            raise ToyotaApiError(error_message)
```

Reply:
Looking at the HTTPX response structure it returns both the status code and associated text. I've used that.